src/infrastructure/YamlGenericTemplateStorage.go

```diff
@@ -0,0 +1,526 @@
+package infrastructure
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"rol/app/interfaces"
+	"rol/domain"
+	"runtime"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+//YamlGenericTemplateStorage is a storage for yaml templates
+type YamlGenericTemplateStorage[TemplateType domain.DeviceTemplate] struct {
+	//TemplatesDirectory is a directory where the templates are located
+	TemplatesDirectory string
+	logger             *logrus.Logger
+	logSourceName      string
+}
+
+//QueryUnit represents bracketed expression that is part of query string
+type QueryUnit struct {
+	FieldName  string
+	Comparator string
+	ValueIndex int
+}
+
+//NewYamlGenericTemplateStorage is a constructor for YamlGenericTemplateStorage
+//
+//Params:
+//	dirName - directory name where the templates are located
+//	log - logrus.Logger
+func NewYamlGenericTemplateStorage[TemplateType domain.DeviceTemplate](dirName string, log *logrus.Logger) *YamlGenericTemplateStorage[TemplateType] {
+	model := new(TemplateType)
+	_, b, _, _ := runtime.Caller(0)
+	rootPath := filepath.Join(filepath.Dir(b), "../")
+
+	templatesDirectory := path.Join(rootPath, dirName)
+	return &YamlGenericTemplateStorage[TemplateType]{
+		TemplatesDirectory: templatesDirectory,
+		logger:             log,
+		logSourceName:      fmt.Sprintf("YamlGenericTemplateStorage<%s>", reflect.TypeOf(*model).Name()),
+	}
+}
+
+func (y *YamlGenericTemplateStorage[TemplateType]) getTemplateObjFromYaml(templateName string) (*TemplateType, error) {
```

Comment:
I've seen deviceTest.yml (roughly) come here in tests, but the service doesn't need to know anything about the data storage format.
We don't have to show implementation details.